modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/programfile/ProgramFileWriter.java

```diff
@@ -108,8 +108,12 @@ private static void writeCP(DataOutputStream dataOutStream,
                 case CP_ENTRY_UTF8:
                     String stringVal = ((UTF8CPEntry) cpEntry).getValue();
                     byte[] bytes = toUTF(stringVal);
-                    dataOutStream.writeShort(bytes.length);
-                    dataOutStream.writeUTF(stringVal);
+                    if (stringVal != null) {
+                        dataOutStream.writeShort(bytes.length);
+                        dataOutStream.writeUTF(stringVal);
```

Comment:
We already have the UTF string value as bytes, in "bytes", so we should just do writeBytes here.